jwql/website/apps/jwql/api_views.py

```diff
@@ -0,0 +1,268 @@
+from .data_containers import get_filenames_by_proposal
+from .data_containers import get_filenames_by_rootname
+from .data_containers import get_instrument_proposals
+from .data_containers import get_preview_images_by_instrument
+from .data_containers import get_preview_images_by_proposal
+from .data_containers import get_preview_images_by_rootname
+from .data_containers import get_thumbnails_by_instrument
+from .data_containers import get_thumbnails_by_proposal
+from .data_containers import get_thumbnails_by_rootname
+
+
+def all_proposals(request):
+    """Return a list of proposals for the mission
+
+    Parameters
+    ----------
+    request : HttpRequest object
+        Incoming request from the webpage
+
+    Returns
+    -------
+    JsonResponse object
+        Outgoing response sent to the webpage
+    """
+
+    proposals = get_all_proposals()
+    return JsonResponse({'proposals': proposals})
+
+
+def filenames_by_proposal(request, proposal):
+    """Return a list of filenames for the given ``proposal``
+
+    Parameters
+    ----------
+    request : HttpRequest object
+        Incoming request from the webpage
+    proposal : str
+        The five-digit proposal number (e.g. ``88600``)
+
+    Returns
+    -------
+    JsonResponse object
+        Outgoing response sent to the webpage
+    """
+
+    filenames = get_filenames_by_proposal(proposal)
+    return JsonResponse({'filenames': filenames})
+
+
+def filenames_by_rootname(request, rootname):
+    """Return a list of filenames for the given ``rootname``
+
+    Parameters
+    ----------
+    request : HttpRequest object
+        Incoming request from the webpage
+    rootname : str
+        The rootname of interest (e.g. ``jw86600008001_02101_00007_guider2``).
+
+    Returns
+    -------
+    JsonResponse object
+        Outgoing response sent to the webpage
+    """
+
+    filenames = get_filenames_by_rootname(rootname)
+    return JsonResponse({'filenames': filenames})
+
+
+def instrument_proposals(request, inst):
+    """Return a list of proposals for the given instrument
+
+    Parameters
+    ----------
+    request : HttpRequest object
+        Incoming request from the webpage
+    inst : str
+        Name of JWST instrument
+
+    Returns
+    -------
+    JsonResponse object
+        Outgoing response sent to the webpage
+    """
+
+    proposals = get_instrument_proposals(inst)
+    return JsonResponse({'proposals': proposals})
+
+
+def preview_images_by_instrument(request, inst):
+    """Return a list of available preview images in the filesystem for
+    the given instrument.
+
+    Parameters
+    ----------
+    request : HttpRequest object
+        Incoming request from the webpage
+    instrument : str
+        The instrument of interest.  The name of the instrument must
+        mach one of the following: (``nircam``, ``NIRCam``, ``niriss``,
+        ``NIRISS``, ``nirspec``, ``NIRSpec``, ``miri``, ``MIRI``,
+        ``fgs``, ``FGS``).
```

Comment:
Two small things: you specify the cases of the instruments in this docstring, but not in other views that also include an instrument argument (I think consistency would be better); and the docstring calls the argument `instrument` when it is just `inst`.